### PR TITLE
Add save as json option to load generator results

### PR DESCRIPTION
**Pull request by srinivashegde86**
This will provide an option to perf tests to save the results in the artifacts directory as JSON. We will be using this later for benchmarking.

### CODE REVIEW

File: shared/loadgenerator/loadgenerator.go [hunk 1]
```diff
@@ -77,3 +82,30 @@ func (g *GeneratorOptions) RunLoadTest(resolvableDomain bool) (*GeneratorResults
...
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+	json, err := json.Marshal(gr)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(json)
```
Comment (adrcunha) [inline shared/loadgenerator/loadgenerator.go]: Merge with the line below? (holding so you can act on this if you want)

/lgtm
/hold

Comment (srinivashegde86) [reply shared/loadgenerator/loadgenerator.go]: done

**Pull request merged**